### DISCUSSION TITLE
ID field generation algorithm

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#131] [Ideas]
Usually, we don't want to expose the default auto-increment id of the database, so that front-end users may obtain some information, such as the number of users.

One approach is to use id encryption to generate a new id, which requires encryption and decryption for each input and output, consuming computing performance. Another way is to store one more complex id, which consumes physical memory.

An ideal id should preferably meet the following conditions:
1. Global uniqueness, unguessable
2. Incremental
3. High availability and high performance

Therefore, we can use the snowflake algorithm to directly store an id field.

@wu-clan What do you think? Any ideas?

**Comment by wu-clan** [comment]
In addition, I don‘t think there seems to be any code change when using Snowflake ID. 

Only need to modify the general ID configuration.

**Reply by wu-clan** [reply]
Hi, @downdawn 

I don't think we need to implement the snowflake algorithm here, maybe add a document that explains how to use

**Reply by downdawn** [reply]
Agreed.

The Snowflake algorithm is not mandatory; we can explain how to customize the id_key.
